vortex-ffi/src/error.rs

```diff
@@ -0,0 +1,44 @@
+use std::ffi::{c_char, c_int};
+use std::ptr;
+
+use vortex::error::VortexResult;
+
+#[repr(C)]
+pub struct FFIError {
+    pub code: c_int,
+    pub message: *const c_char,
+}
+
+pub unsafe fn into_return_mut<T, V>(
```

Comment:
Not super sure about the func name: `into_return_mut`.

Reply:
many ideas, went for into_return for now